visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -15,15 +16,12 @@ export class CodeChartaService {
 	public loadFiles(nameDataPairs: NameDataPair[]): Promise<void> {
 		return new Promise((resolve, reject) => {
 			nameDataPairs.forEach((nameDataPair: NameDataPair) => {
-				const validationResult: CCValidationResult = FileValidator.validate(nameDataPair.content)
-				if (validationResult.error.length === 0) {
-					const ccFile = this.getCCFile(nameDataPair.fileName, nameDataPair.content)
-					NodeDecorator.preDecorateFile(ccFile)
-					this.storeService.dispatch(addFile(ccFile))
-					if (validationResult.warning.length !== 0) {
-						reject(validationResult)
-					}
-				} else {
+				const validationResult: CCValidationResult = validate(nameDataPair.content)
+				if (validationResult.error.length > 0) {
+					reject(validationResult)
+				}
+				this.addFile(nameDataPair.fileName, nameDataPair.content)
+				if (validationResult.warning.length > 0) {
```

Comment:
I suggest to "emit" warnings or to handle them as errors. Right now errors and warnings are handled identically in this code. Do we ever handle them differently? And if so, why does a warning result in the same behavior here?

Reply:
I was confused by this as well. I think the intention here is to reject errors and display them. We want to resolve and add the file to our state on warnings but still display the warnings.

I committed something that should make this intention clear and also resolves your other discussion regarding returning a validationResult @BridgeAR